MapboxSearch/base/src/main/java/com/mapbox/search/base/location/LocationEngineAdapter.kt

```diff
@@ -19,6 +19,11 @@ import kotlinx.coroutines.Job
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 
+object LocationObservationProperties {
+    @Volatile
+    var locationObservationTimeout: Long? = 1_000L
+}
+
```

Comment:
personally i'd put this as apart of the constructor so i don't have to worry about synchronization issues or calls being made out of sequence.

Reply:
This is also a reference to @DzmitryFomchyn's concern, if it is apart of the constructor, you don't have to worry about changing it under other's noses:
>Furthermore, it's error-prone; if multiple instances of AddressAutofill are created with different locationObservationTimeout values, only the latest one will be utilized.

Reply:
Can you elaborate?

The first option is to provide a global variable that can be set to any value.
The second option is more explicit but requires to pass a new almost irrelevant parameter to every constructor that deals with asynchronous location updates.

Reply:
I don't see it being supplied in a constructor any where?

Reply:
it's only used by the `LocationEngineAdapter`, why not add it as optional to the constructor:
```
class LocationEngineAdapter(
    private val app: Application,
    private val locationProvider: LocationProvider?,
    private val timeProvider: TimeProvider = LocalTimeProvider(),
    private val locationPermissionChecker: (Application) -> Boolean = {
        PermissionsManager.areLocationPermissionsGranted(app)
    },
    private val locationObservationTimeout: Long? = 1_000L
)
```